pkg/control/v2/server/server_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package server
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gotest.tools/assert"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator/state"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/info"
+	"github.com/elastic/elastic-agent/pkg/control/v2/cproto"
+)
+
+func TestFleetStateMapping(t *testing.T) {
```

Comment:
It can be done of course, I wanted to merge the fix quickly so that the backport could land on the 8.8 branch before the final BC. Will give it a shot and see how long it takes

Reply:
Maybe don't cover all the fields, but at least the other state fields, skipping `Components` for now.

Reply:
Erm, I am reading only now your comment, I managed to cover all the relevant attributes except the AgentInfo (that is tricky to create without a DiskStore). I didn't add much variety but I avoided all the zero values so if we are missing some fields we should be able to see that